From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 23 Mar 2011 03:53:50 +0000
Subject: [PATCH] r8169: Fix up backport of "r8169: keep firmware in memory."

In 2.6.32, the first parameter of mdio_read() in r8169 is the device
MMIO address, not the address of the driver context.
---
 drivers/net/r8169.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 350880f..1cbf195 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -1907,7 +1907,7 @@ static void rtl8168d_1_hw_phy_config(struct rtl8169_private *tp)
 
 	mdio_write(ioaddr, 0x1f, 0x0005);
 	mdio_write(ioaddr, 0x05, 0x001b);
-	if ((mdio_read(tp, 0x06) != 0xbf00) ||
+	if ((mdio_read(ioaddr, 0x06) != 0xbf00) ||
 	    (rtl_apply_firmware(tp, FIRMWARE_8168D_1) < 0)) {
 		if (netif_msg_probe(tp))
 			printk(KERN_WARNING "%s: unable to apply firmware patch\n",
@@ -2014,7 +2014,7 @@ static void rtl8168d_2_hw_phy_config(struct rtl8169_private *tp)
 
 	mdio_write(ioaddr, 0x1f, 0x0005);
 	mdio_write(ioaddr, 0x05, 0x001b);
-	if ((mdio_read(tp, 0x06) != 0xb300) ||
+	if ((mdio_read(ioaddr, 0x06) != 0xb300) ||
 	    (rtl_apply_firmware(tp, FIRMWARE_8168D_2) < 0)) {
 		if (netif_msg_probe(tp))
 			printk(KERN_WARNING "%s: unable to apply firmware patch\n",
-- 
1.7.4.1

